src/fondant/data_io.py

```diff
@@ -124,11 +128,16 @@ def load_dataframe(self) -> dd.DataFrame:
                 as well as the index columns.
         """
         # load index into dataframe
+        logging.info(f"Loading index...")
         dataframe = self._load_index()
+
         for name, subset in self.component_spec.consumes.items():
             fields = list(subset.fields.keys())
             subset_df = self._load_subset(name, fields)
             # left joins -> filter on index
+            # make sure that dataframe has same number of partitions
+            # as subset
+            dataframe = dataframe.repartition(npartitions=subset_df.npartitions)
```

Comment:
I'll have a look at this, eventually we can maybe even flag it as an issue to the `Dask` repo